src/test/registration.test.tsx

```diff
@@ -0,0 +1,40 @@
+// registration.test.tsx
+import { fireEvent, render, waitFor } from '@testing-library/react';
+import { describe, expect, it } from 'vitest';
+import { Registration } from '../pages/registration/registration';
+
+describe('Registration Page', () => {
+  it('renders registration form', () => {
+    const { getByText } = render(<Registration />);
+    expect(getByText('Register')).toBeInTheDocument();
+  });
+
+  it('validates email input', () => {
+    const { getByPlaceholderText, getByText } = render(<Registration />);
+    const emailInput = getByPlaceholderText('Email');
+    fireEvent.change(emailInput, { target: { value: 'invalid-email' } });
+    expect(getByText('Invalid email address')).toBeInTheDocument();
+  });
+
+  it('validates password input', () => {
+    const { getByPlaceholderText, getByText } = render(<Registration />);
+    const passwordInput = getByPlaceholderText('Password');
+    fireEvent.change(passwordInput, { target: { value: 'short' } });
+    expect(getByText('Password must be at least 8 characters')).toBeInTheDocument();
```

Comment:
```suggestion
    expect(await findByText(/Password must be at least 8 characters/)).toBeInTheDocument();
```

Updated the rule – now we're checking for a partial match in the message instead of an exact match.

Reply:
Good idea